agent/fake_api_server_test.go

```diff
@@ -109,6 +109,8 @@ func (fs *FakeAPIServer) AddJob(env map[string]string) *FakeJob {
 }
 
 func (fs *FakeAPIServer) Assign(agent *FakeAgent, job *FakeJob) {
+	fs.mu.Lock()
+	defer fs.mu.Unlock()
```

Comment:
Weirdly, this seems to have caused a bunch of 60-second timeouts… deadlocks or something?